crypto/merkletrie/committer_test.go

```diff
@@ -187,3 +193,51 @@ func TestMultipleCommits(t *testing.T) {
 	}
 	require.Equal(t, storageSize1, storageSize2)
 }
+
+func TestIterativeCommits(t *testing.T) {
+	testSize := 1000
+
+	memConfig := MemoryConfig{
+		NodesCountPerPage:         116,
+		CachedNodesCount:          9000,
+		PageFillFactor:            0.95,
+		MaxChildrenPagesThreshold: 64,
+	}
+
+	hashes := make([]crypto.Digest, testSize)
+	for i := 0; i < len(hashes); i++ {
+		hashes[i] = crypto.Hash([]byte{byte(i), byte(i >> 8), byte(i >> 16), byte(i >> 24), byte(0), byte(0)})
+	}
+
+	// initialize memory container.
+	mc := &InMemoryCommitter{}
+	mt, _ := MakeTrie(mc, memConfig)
+	for i := 0; i < len(hashes); i++ {
+		added, err := mt.Add(hashes[i][:])
+		require.True(t, added)
+		require.NoError(t, err)
+	}
+	_, err := mt.Commit()
+	require.NoError(t, err)
+
+	for r := 0; r < 100; r++ {
+		newMC := mc.Duplicate(true)
+		mt, _ = MakeTrie(newMC, memConfig)
+		mc = newMC
```

Comment:
Why mc is duplicated to newMC, and then the original (mc) is discarded?

Reply:
I wanted to ensure that there is no chance of memory slices being shared. It is intended to simulate persisting to disk and reloading from disk.